src/frontend/src/optimizer/rule/stream/split_now_or_rule.rs

```diff
@@ -0,0 +1,92 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use risingwave_common::types::DataType;
+
+use crate::expr::{ExprImpl, ExprType, FunctionCall};
+use crate::optimizer::plan_node::{LogicalFilter, LogicalShare, LogicalUnion, PlanTreeNodeUnary};
+use crate::optimizer::rule::{BoxedRule, Rule};
+use crate::optimizer::PlanRef;
+
+/// Convert `LogicalFilter` with now or others predicates to a `UNION ALL`
+///
+/// Before:
+/// `LogicalFilter`
+///  now() or others
+///        |
+///      Input
+///
+/// After:
+///         `LogicalUnionAll`
+///       /                  \
+/// `LogicalFilter`       `LogicalFilter`
+/// now() & !others           others
+///     |                     |
+///     \                    /
+///         `LogicalShare`
+///               |
+///             Input
+pub struct SplitNowOrRule {}
+impl Rule for SplitNowOrRule {
+    fn apply(&self, plan: PlanRef) -> Option<PlanRef> {
+        let filter: &LogicalFilter = plan.as_logical_filter()?;
+        let input = filter.input();
+
+        if filter.predicate().conjunctions.len() != 1 {
+            return None;
+        }
+
+        let disjunctions = filter.predicate().conjunctions[0].as_or_disjunctions()?;
+
+        // TODO: handle disjuctions with arms more than 2
+        if disjunctions.len() != 2 {
+            return None;
+        }
```

Comment:
Not very hard if we suppose there is only one disjunction with `Now()` ? I think the user can write multiple common disjunction but only one temporal filter disjunction

Reply:
+1. Let's clean the `TODO` in this or next PR, otherwise it's hard to tell users the boundary of this feature.